raft/src/main/java/org/apache/kafka/raft/internals/EpochElection.java

```diff
@@ -0,0 +1,216 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.raft.internals;
+
+import org.apache.kafka.raft.ReplicaKey;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ *  Tracks the votes cast by voters in an election held by a Nominee.
+ */
+public class EpochElection {
+    private Map<Integer, VoterState> voterStates;
+
+    public EpochElection(Set<ReplicaKey> voters) {
+        this.voterStates = voters.stream()
+            .collect(
+                Collectors.toMap(
+                    ReplicaKey::id,
+                    VoterState::new
+                )
+            );
+    }
+
+    /**
+     * Record a vote from a voter.
+     * @param voterId The id of the voter
+     * @param isGranted true if the vote is granted, false if it is rejected
+     * @return true if the voter had not been previously recorded
+     */
+    public boolean recordVote(int voterId, boolean isGranted) {
+        boolean wasUnrecorded = false;
+        VoterState voterState = getVoterStateOrThrow(voterId);
+        if (voterState.state == VoterState.State.UNRECORDED) {
+            wasUnrecorded = true;
+        }
+        if (isGranted) {
+            voterState.setState(VoterState.State.GRANTED);
+        } else {
+            voterState.setState(VoterState.State.REJECTED);
+        }
+        return wasUnrecorded;
+    }
+
+    /**
+     * Returns if a voter has granted the vote.
+     * @param voterId The id of the voter
+     * @throws IllegalArgumentException if the voter is not in the set of voters
+     */
+    public boolean isGrantedVoter(int voterId) {
+        return getVoterStateOrThrow(voterId).state == VoterState.State.GRANTED;
+    }
+
+    /**
+     * Returns if a voter has rejected the vote.
+     * @param voterId The id of the voter
+     * @throws IllegalArgumentException if the voter is not in the set of voters
+     */
+    public boolean isRejectedVoter(int voterId) {
+        return getVoterStateOrThrow(voterId).state == VoterState.State.REJECTED;
+    }
+
+    /**
+     * The set of voter ids.
+     */
+    public Set<Integer> voterIds() {
+        return Collections.unmodifiableSet(voterStates.keySet());
+    }
+
+    /**
+     * Get the collection of voter states.
+     */
+    public Collection<VoterState> voterStates() {
+        return Collections.unmodifiableCollection(voterStates.values());
+    }
```

Comment:
Looks like this doesn't need to be public. Looks like this method can be removed since it is not used.